system/CodeIgniter.php

```diff
@@ -968,6 +969,12 @@ protected function gatherOutput($cacheConfig = null, $returned = null)
 	 */
 	public function storePreviousURL($uri)
 	{
+		// Only valid for some requests
+		if (! $this->request instanceof IncomingRequest || $this->request->isCLI() || $this->request->isAJAX())
```

Comment:
I'm a little concerned with checking against IncomingRequest. it's entirely possible for someone to use a completely different IncomingRequest-like class. While I would think they would extend IncomingRequest there's nothing that limits them from doing something crazy. Better to either not check against a class type and just do the AJAX and CLI checks, or check against Request.

Reply:
That is how I had it originally but since neither **Request** nor **RequestInterface** require `isAJAX()` and `isCLI()` it was causing issues, e.g. CLI requests where giving "method doesn't exist" exceptions.

I guess I will try it by checking for each method, but I'm also open to an additional interface or some other means of defining a "browser visit".